website/documentation/guides/administer-shoots/scalability/_index.md

```diff
@@ -0,0 +1,141 @@
+- Applications interacting with the K8s API: If you know an object by its name, use it. Using label selector queries is expensive as the filtering happens only within the `kube-apiserver` and not `etcd`, hence all resources must first pass completely from `etcd` to `kube-apiserver`.
+- Use (single object) caches within your controllers. Check https://github.com/gardener/gardener/issues/7593 for an example.
+
+### `Nodes`
+
+When talking about the scalability of a Kubernetes cluster, `Nodes` are probably mentioned in the first place... well, obviously not in this guide. While vanilla Kubernetes lists 5000 `Nodes` as its upper limit, pushing that dimension is not feasible. Most clusters should run with fewer than 300 `Nodes`. If you scale beyond that, be extra careful and closely monitor ETCD and `kube-apiserver`.
+
+The scalability of `Nodes` is subject to a range of limiting factors. Some of them can only be defined upon cluster creation and remain immutable during a cluster lifetime. So let's discuss the most important dimensions.
+
+**CIDR**: 
+
+Upon cluster creation, you have to specify or use the default values for several network segments. There are dedicated CIDRs for services, `Pods` and `Nodes`. Each defines a range of IP addresses available for the individual resource type. Obviously, the maximum of possible `Nodes` is capped by the CIDR for `Nodes`. 
+However, there is a second limiting factor, which is the pod CIDR combined with the `nodeCIDRMaskSize`. This mask is used to divide the pod CIDR into smaller subnets, where each blocks gets assigned to a node. With a `/16` pod network and a `/24` nodeCIDRMaskSize a cluster can scale up to 256 `Nodes`. Please check the [documentation](https://gardener.cloud/docs/gardener/usage/shoot_networking/) for details.
+
+Even though a `/24` nodeCIDRMaskSize translates to a theoretical 256 pod IP addresses per `Node`, the `maxPods` setting should be less than 1/2 of this value. This gives the system some breathing room for churn and minimizes the risk for strange effects like mis-routed packages caused by immediate re-use of IPs.
+
+**Cloud provider capacity**:
+
+Most of the time, `Nodes` in Kubernetes translate to virtual machines on a hyperscaler. An attempt to add more `Nodes` to a cluster might fail due to capacity issues resulting in an error message like this:
+
+```
+Cloud provider message - machine codes error: code = [Internal] message = [InsufficientInstanceCapacity: We currently do not have sufficient <instance type> capacity in the Availability Zone you requested. Our system will be working on provisioning additional capacity. 
+```
+
+In heavily utilized regions individual clusters are competing for scarce resources. So before choosing a region / zone, try to ensure the hyperscaler supports your anticipated growth. This might be done through quota requests or by contacting the respective support teams.
```

Comment:
```suggestion
In heavily utilized regions, individual clusters are competing for scarce resources. So before choosing a region / zone, try to ensure that the hyperscaler supports your anticipated growth. This might be done through quota requests or by contacting the respective support teams.
```